src/index.js

```diff
@@ -72,8 +72,7 @@ export default function validate(rules) {
 
       const value = scope === 'params'
         ? getValueByPath(ctx.params, name)
-        : scope === 'query'
-          ? getValueByPath(ctx.request.query, name)
+        : ctx.request[`${scope}`] ? getValueByPath(ctx.request[`${scope}`], name)
           : (ctx.request.body && getValueByPath(ctx.request.body, name))
```

Comment:
Is this necessary anymore? It seems to me the whole thing could be rewritten as:

```
const data = ctx[scope] || ctx.request[scope];
const value = getValueAtPath(data, path); // note I rename 'name' to 'path' and the method name to be clearer
```